Source/Public/New-DocusaurusHelp.ps1

```diff
@@ -167,18 +175,59 @@ function New-DocusaurusHelp() {
 
-    # make sure the passed module is valid
-    if (Test-Path($Module)) {
-        Import-Module $Module -Force -Global
-        $Module = [System.IO.Path]::GetFileNameWithoutExtension($Module)
+    # Get the module's name fromn the supplied markdown files.
+    if ($PSCmdlet.ParameterSetName.Equals('PlatyPSMarkdownPath'))
+    {
+        # Get the module's name from the supplied markdown files.
+        $moduleName = Get-ChildItem -LiteralPath $PlatyPSMarkdownPath -Filter *.md |
+            ForEach-Object { [System.IO.File]::ReadAllLines($_.FullName) -match '^Module Name: ' } |
+            Select-Object -Unique
+
+        # Throw if null or we've got more than one item.
+        if ($null -eq $moduleName)
+        {
+            $errSentence1 = 'Unable to determine the module name from the supplied markdown files.'
+            $errSentence2 = 'Please confirm their validity and try again.'
+            $PSCmdlet.ThrowTerminatingError([System.Management.Automation.ErrorRecord]::new(
+                    [System.ArgumentException]::new("$errSentence1 $errSentence2", 'PlatyPSMarkdownPath'),
+                    'ModuleNameIndeterminateError',
+                    [System.Management.Automation.ErrorCategory]::InvalidResult,
+                    $moduleName
+                ))
+        }
+        elseif ($moduleName -isnot [System.String])
+        {
+            $errSentence1 = "More than one module name was found within the supplied markdown files ('$([System.String]::Join("', '", $moduleName))')."
+            $errSentence2 = 'Please supply unique markdown files for a single module and try again.'
+            $PSCmdlet.ThrowTerminatingError([System.Management.Automation.ErrorRecord]::new(
+                    [System.ArgumentException]::new("$errSentence1 $errSentence2", 'PlatyPSMarkdownPath'),
+                    'DuplicateModuleNameError',
+                    [System.Management.Automation.ErrorCategory]::InvalidResult,
+                    $moduleName
+                ))
+        }
+
+        # Trim off the leading characters before continuing.
+        $Module = $moduleName = $moduleName -replace '^Module Name: '
     }
 
-    if (-Not(Get-Module -Name $Module)) {
-        $Module = $Module
-        throw "New-DocusaurusHelp: Specified module '$Module' is not loaded"
+    # make sure the passed module is valid
+    if (!(Get-Module -Name $Module))
+    {
+        if (Test-Path -LiteralPath $Module) {
+            Import-Module $Module -Force -Global
+            $Module = [System.IO.Path]::GetFileNameWithoutExtension($Module)
+        }
+        else {
+            $Module = $Module
+            throw "New-DocusaurusHelp: Specified module '$Module' is not loaded"
+        }
     }
 
-    $moduleName = [io.path]::GetFileName($module)
+    if (!(Test-Path -LiteralPath Microsoft.PowerShell.Core\Variable::moduleName))
+    {
+        $moduleName = [io.path]::GetFileName($module)
+    }
```

Comment:
Do we need this when module name is parsed from markdown already?

For the original flow this could be rewritten to use `Import-Module -PassThru` which returns ModuleInfo with a name property.

Reply:
Apologies for this, I was clearly more tired than I thought last night. This too has been addressed.